mmengine/visualization/vis_backend.py

```diff
@@ -610,3 +611,180 @@ def close(self):
+        >>> vis_backend.add_config(cfg)
+
+    Args:
+        save_dir (str): The root directory to save the files
+            produced by the backend.
+        exp_name (str, optional): The experiment name. Default to None.
+        run_name (str, optional): The run name. Default to None.
+        tags (dict, optional): The tags to be added to the experiment.
+            Default to None.
+        params (dict, optional): The params to be added to the experiment.
+            Default to None.
+        tracking_uri (str, optional): The tracking uri. Default to None.
+        log_artifact (bool, optional): Whether to log the artifact.
+            Default to False.
+    """
+
+    def __init__(self,
+                 save_dir: str,
+                 exp_name: Optional[str] = None,
+                 run_name: Optional[str] = None,
+                 tags: Optional[dict] = None,
+                 params: Optional[dict] = None,
+                 tracking_uri: Optional[str] = None,
+                 log_artifact: bool = False):
+        super().__init__(save_dir)
+        self._exp_name = exp_name
+        self._run_name = run_name
+        self._tags = tags
+        self._params = params
+        self._tracking_uri = tracking_uri
+        self._log_artifact = log_artifact
+
+    def _init_env(self):
+        """Setup env for MLflow."""
+        if not os.path.exists(self._save_dir):
+            os.makedirs(self._save_dir, exist_ok=True)  # type: ignore
+
+        try:
+            import mlflow
+        except ImportError:
+            raise ImportError(
+                'Please run "pip install mlflow" to install mlflow'
+            )  # type: ignore
+        self._mlflow = mlflow
+
+        if self._tracking_uri is not None:
+            self._mlflow.set_tracking_uri(self._tracking_uri)
+        else:
+            self._mlflow.set_tracking_uri(
+                f'file:\\{os.path.abspath(self._save_dir)}')
+
+        self._exp_name = self._exp_name or 'Default'
+
+        if self._mlflow.get_experiment_by_name(self._exp_name) is None:
+            self._mlflow.create_experiment(self._exp_name)
+
+        self._mlflow.set_experiment(self._exp_name)
+
+        if self._run_name is not None:
+            self._mlflow.set_tag('mlflow.runName', self._run_name)
+        if self._tags is not None:
+            self._mlflow.set_tags(self._tags)
+        if self._params is not None:
+            self._mlflow.log_params(self._params)
+
+    @property  # type: ignore
+    @force_init_env
+    def experiment(self):
+        """Return MLflow object."""
+        return self._mlflow
+
+    @force_init_env
+    def add_config(self, config: Config, **kwargs) -> None:
+        """Record the config to mlflow.
+
+        Args:
+            config (Config): The Config object
+        """
+        self.cfg = config
+        self._mlflow.log_params(self._flatten(self.cfg))
+        self._mlflow.log_text(self.cfg.pretty_text, 'config.py')
```

Comment:
I tested this backend in mmcls and found that the above parameters `self._flatten(self.cfg)` were recorded but `config.py` was not.

Reply:
Hi @sh0622-kim , it seems the `config.py` was not uploaded.

Reply:
can you tell me the contents of config.py executed in mmcls?
I confirmed that it is saved in the environment I made, but I will check again what the problem is.

Reply:
The following is my config
```python
# set visualizer
vis_backends = [dict(type='MLflowVisBackend', tracking_uri='http://localhost:5000', log_artifact=True)]
visualizer = dict(type='ClsVisualizer', vis_backends=vis_backends)
```

Nothing was uploaded.

![image](https://user-images.githubusercontent.com/58739961/221510258-17b70ef1-6c78-4195-b969-548cf5ae7638.png)

Reply:
@zhouzaida 

I am struggling to reproduce this issue. 
I attach the `config.py`, I wrote to reproduce this phenomenon.
Can someone please confirm if I'm missing something?

```python
model = dict(
    type='ImageClassifier',
    backbone=dict(
        type='ResNet_CIFAR',
        depth=18,
        num_stages=4,
        out_indices=(3, ),
        style='pytorch'),
    neck=dict(type='GlobalAveragePooling'),
    head=dict(
        type='LinearClsHead',
        num_classes=10,
        in_channels=512,
        loss=dict(type='CrossEntropyLoss', loss_weight=1.0)))
dataset_type = 'CIFAR10'
data_preprocessor = dict(
    num_classes=10,
    mean=[125.307, 122.961, 113.8575],
    std=[51.5865, 50.847, 51.255],
    to_rgb=False)
train_pipeline = [
    dict(type='RandomCrop', crop_size=32, padding=4),
    dict(type='RandomFlip', prob=0.5, direction='horizontal'),
    dict(type='PackClsInputs')
]
test_pipeline = [dict(type='PackClsInputs')]
train_dataloader = dict(
    batch_size=16,
    num_workers=2,
    dataset=dict(
        type='CIFAR10',
        data_prefix='data/cifar10',
        test_mode=False,
        pipeline=[
            dict(type='RandomCrop', crop_size=32, padding=4),
            dict(type='RandomFlip', prob=0.5, direction='horizontal'),
            dict(type='PackClsInputs')
        ]),
    sampler=dict(type='DefaultSampler', shuffle=True))
val_dataloader = dict(
    batch_size=16,
    num_workers=2,
    dataset=dict(
        type='CIFAR10',
        data_prefix='data/cifar10/',
        test_mode=True,
        pipeline=[dict(type='PackClsInputs')]),
    sampler=dict(type='DefaultSampler', shuffle=False))
val_evaluator = dict(type='Accuracy', topk=(1, ))
test_dataloader = dict(
    batch_size=16,
    num_workers=2,
    dataset=dict(
        type='CIFAR10',
        data_prefix='data/cifar10/',
        test_mode=True,
        pipeline=[dict(type='PackClsInputs')]),
    sampler=dict(type='DefaultSampler', shuffle=False))
test_evaluator = dict(type='Accuracy', topk=(1, ))
optim_wrapper = dict(
    optimizer=dict(type='SGD', lr=0.1, momentum=0.9, weight_decay=0.0001))
param_scheduler = dict(
    type='MultiStepLR', by_epoch=True, milestones=[100, 150], gamma=0.1)
train_cfg = dict(by_epoch=True, max_epochs=200, val_interval=1)
val_cfg = dict()
test_cfg = dict()
auto_scale_lr = dict(base_batch_size=128)
default_scope = 'mmcls'
default_hooks = dict(
    timer=dict(type='IterTimerHook'),
    logger=dict(type='LoggerHook', interval=100),
    param_scheduler=dict(type='ParamSchedulerHook'),
    checkpoint=dict(type='CheckpointHook', interval=1),
    sampler_seed=dict(type='DistSamplerSeedHook'),
    visualization=dict(type='VisualizationHook', enable=False))
env_cfg = dict(
    cudnn_benchmark=False,
    mp_cfg=dict(mp_start_method='fork', opencv_num_threads=0),
    dist_cfg=dict(backend='nccl'))
# set visualizer
vis_backends = [dict(type='MLflowVisBackend',
                     tracking_uri='http://localhost:5000', log_artifact=True)]
visualizer = dict(type='ClsVisualizer', vis_backends=vis_backends)
log_level = 'INFO'
load_from = None
resume = False
randomness = dict(seed=None, deterministic=False)

```
![image](https://user-images.githubusercontent.com/66167254/222953808-90af2469-e8f4-4b49-b8aa-7fd4dfc1c2b2.png)

Reply:
OK. I will have a try.